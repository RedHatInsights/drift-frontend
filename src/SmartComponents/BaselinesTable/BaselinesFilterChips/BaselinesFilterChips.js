import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Chip, ChipGroup, ChipGroupToolbarItem } from '@patternfly/react-core';

class BaselinesFilterChips extends Component {
    constructor(props) {
        super(props);
        this.state = {
            chipGroup: [
                { category: 'Baseline name', chips: [ this.props.nameSearch ]}
            ]
        };
    }

    componentDidUpdate(prevProps) {
        if (this.props.nameSearch !== prevProps.nameSearch) {
            let newChipGroup = [
                { category: 'Baseline name', chips: [ this.props.nameSearch ]}
            ];

            this.setState({ chipGroup: newChipGroup });
        }
    }

    render() {
        const { chipGroup } = this.state;
        const { clearTextFilter } = this.props;

        return (
            <ChipGroup withToolbar>
                { chipGroup.map(currentGroup => (
                    <ChipGroupToolbarItem key={ currentGroup.category } categoryName={ currentGroup.category }>
                        { currentGroup.chips.map(chip => (
                            <Chip key={ chip } onClick={ clearTextFilter }>
                                { chip }
                            </Chip>
                        )) }
                    </ChipGroupToolbarItem>
                )) }
            </ChipGroup>
        );
    }
}

BaselinesFilterChips.propTypes = {
    nameSearch: PropTypes.string,
    clearTextFilter: PropTypes.func
};

export default BaselinesFilterChips;
