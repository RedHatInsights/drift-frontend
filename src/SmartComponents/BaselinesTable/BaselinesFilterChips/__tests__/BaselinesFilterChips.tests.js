import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import BaselinesFilterChips from '../BaselinesFilterChips';

describe('BaselinesFilterChips', () => {
    it('should render shallow correctly', () =>{
        const wrapper = shallow(
            <BaselinesFilterChips />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render mount correctly', () =>{
        const wrapper = mount(
            <BaselinesFilterChips />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render chip correctly', () =>{
        const props = {
            nameSearch: 'something'
        };
        const wrapper = shallow(
            <BaselinesFilterChips { ...props }/>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render chip update correctly', () =>{
        const props = {
            nameSearch: 'something'
        };
        const wrapper = shallow(
            <BaselinesFilterChips { ...props }/>
        );

        wrapper.setProps({ nameSearch: 'something else' });

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    describe('API', () => {
        it('should have call clearTextFilter', () => {
            const clearTextFilter = jest.fn();
            const wrapper = mount(<BaselinesFilterChips
                nameSearch={ 'something' }
                clearTextFilter={ clearTextFilter }
            />);

            wrapper.find('.pf-c-button').simulate('click');
            expect(clearTextFilter).toHaveBeenCalledTimes(1);
        });
    });
});
