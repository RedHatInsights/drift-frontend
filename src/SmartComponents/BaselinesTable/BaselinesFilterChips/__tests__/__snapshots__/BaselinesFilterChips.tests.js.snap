// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaselinesFilterChips should render chip correctly 1`] = `
<ChipGroup
  className=""
  collapsedText="\${remaining} more"
  defaultIsOpen={false}
  expandedText="Show Less"
  numChips={3}
  withToolbar={true}
>
  <ChipGroupToolbarItem
    categoryName="Baseline name"
    key="Baseline name"
  >
    <Component
      key="something"
    >
      something
    </Component>
  </ChipGroupToolbarItem>
</ChipGroup>
`;

exports[`BaselinesFilterChips should render chip update correctly 1`] = `
<ChipGroup
  className=""
  collapsedText="\${remaining} more"
  defaultIsOpen={false}
  expandedText="Show Less"
  numChips={3}
  withToolbar={true}
>
  <ChipGroupToolbarItem
    categoryName="Baseline name"
    key="Baseline name"
  >
    <Component
      key="something else"
    >
      something else
    </Component>
  </ChipGroupToolbarItem>
</ChipGroup>
`;

exports[`BaselinesFilterChips should render mount correctly 1`] = `
<BaselinesFilterChips>
  <ChipGroup
    className=""
    collapsedText="\${remaining} more"
    defaultIsOpen={false}
    expandedText="Show Less"
    numChips={3}
    withToolbar={true}
  >
    <InnerChipGroup
      className=""
      collapsedText="\${remaining} more"
      defaultIsOpen={false}
      expandedText="Show Less"
      isOpen={false}
      numChips={3}
      onToggleCollapse={[Function]}
      withToolbar={true}
    >
      <ChipGroupToolbarItem
        categoryName="Baseline name"
        key=".$Baseline name"
      >
        <GenerateId
          prefix="pf-random-id-"
        >
          <ul
            className="pf-c-chip-group pf-m-toolbar"
          >
            <li>
              <h4
                className="pf-c-chip-group__label"
                id="pf-random-id-0"
              >
                Baseline name
              </h4>
              <ul
                className="pf-c-chip-group"
              >
                <Component
                  component="li"
                  key=".0"
                >
                  <ComponentWithOuia
                    component={[Function]}
                    componentProps={
                      Object {
                        "children": undefined,
                        "component": "li",
                        "onClick": undefined,
                      }
                    }
                    consumerContext={null}
                  >
                    <Chip
                      className=""
                      closeBtnAriaLabel="close"
                      component="li"
                      isOverflowChip={false}
                      isReadOnly={false}
                      onClick={[Function]}
                      ouiaContext={
                        Object {
                          "isOuia": false,
                          "ouiaId": null,
                        }
                      }
                      tooltipPosition="top"
                    >
                      <GenerateId
                        prefix="pf-random-id-"
                      >
                        <li
                          className="pf-c-chip"
                        >
                          <span
                            className="pf-c-chip__text"
                            id="pf-random-id-1"
                          />
                          <ChipButton
                            aria-labelledby="remove_pf-random-id-1 pf-random-id-1"
                            ariaLabel="close"
                            id="remove_pf-random-id-1"
                            onClick={[Function]}
                          >
                            <Component
                              aria-label="close"
                              aria-labelledby="remove_pf-random-id-1 pf-random-id-1"
                              className=""
                              id="remove_pf-random-id-1"
                              onClick={[Function]}
                              variant="plain"
                            >
                              <ComponentWithOuia
                                component={[Function]}
                                componentProps={
                                  Object {
                                    "aria-label": "close",
                                    "aria-labelledby": "remove_pf-random-id-1 pf-random-id-1",
                                    "children": <TimesCircleIcon
                                      aria-hidden="true"
                                      color="currentColor"
                                      noVerticalAlign={false}
                                      size="sm"
                                      title={null}
                                    />,
                                    "className": "",
                                    "id": "remove_pf-random-id-1",
                                    "onClick": [Function],
                                    "variant": "plain",
                                  }
                                }
                                consumerContext={
                                  Object {
                                    "isOuia": false,
                                    "ouiaId": null,
                                  }
                                }
                              >
                                <Button
                                  aria-label="close"
                                  aria-labelledby="remove_pf-random-id-1 pf-random-id-1"
                                  className=""
                                  id="remove_pf-random-id-1"
                                  onClick={[Function]}
                                  ouiaContext={
                                    Object {
                                      "isOuia": false,
                                      "ouiaId": null,
                                    }
                                  }
                                  variant="plain"
                                >
                                  <button
                                    aria-disabled={null}
                                    aria-label="close"
                                    aria-labelledby="remove_pf-random-id-1 pf-random-id-1"
                                    className="pf-c-button pf-m-plain"
                                    disabled={false}
                                    id="remove_pf-random-id-1"
                                    onClick={[Function]}
                                    tabIndex={null}
                                    type="button"
                                  >
                                    <TimesCircleIcon
                                      aria-hidden="true"
                                      color="currentColor"
                                      noVerticalAlign={false}
                                      size="sm"
                                      title={null}
                                    >
                                      <svg
                                        aria-hidden="true"
                                        aria-labelledby={null}
                                        fill="currentColor"
                                        height="1em"
                                        role="img"
                                        style={
                                          Object {
                                            "verticalAlign": "-0.125em",
                                          }
                                        }
                                        viewBox="0 0 512 512"
                                        width="1em"
                                      >
                                        <path
                                          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                                          transform=""
                                        />
                                      </svg>
                                    </TimesCircleIcon>
                                  </button>
                                </Button>
                              </ComponentWithOuia>
                            </Component>
                          </ChipButton>
                        </li>
                      </GenerateId>
                    </Chip>
                  </ComponentWithOuia>
                </Component>
              </ul>
            </li>
          </ul>
        </GenerateId>
      </ChipGroupToolbarItem>
    </InnerChipGroup>
  </ChipGroup>
</BaselinesFilterChips>
`;

exports[`BaselinesFilterChips should render shallow correctly 1`] = `
<ChipGroup
  className=""
  collapsedText="\${remaining} more"
  defaultIsOpen={false}
  expandedText="Show Less"
  numChips={3}
  withToolbar={true}
>
  <ChipGroupToolbarItem
    categoryName="Baseline name"
    key="Baseline name"
  >
    <Component />
  </ChipGroupToolbarItem>
</ChipGroup>
`;
